src/taskgraph/docker.py

```diff
@@ -237,3 +239,59 @@ def download_and_modify_image():
         raise Exception("No repositories file found!")
 
     return info
+
+
+def load_task(task_id, remove=False):
+    task_def = get_task_definition(task_id)
+
+    if (
+        impl := task_def.get("tags", {}).get("worker-implementation")
+    ) != "docker-worker":
+        print(f"Tasks with worker-implementation '{impl}' are not supported!")
+        return 1
+
+    command = task_def["payload"].get("command")
+    if not command or not command[0].endswith("run-task"):
+        print("Only tasks using `run-task` are supported!")
+        return 1
+
+    # Remove the payload section of the task's command. This way run-task will
+    # set up the task (clone repos, download fetches, etc) but won't actually
+    # start the core of the task. Instead we'll drop the user into a command
+    # shell.
+    if (index := command.index("--")) > -1:
+        command[index + 1 :] = ["echo 'Task setup complete!'"]
+
+    image_task_id = task_def["payload"]["image"]["taskId"]
+    image_tag = load_image_by_task_id(image_task_id, None)
+
+    env = task_def["payload"].get("env")
+
+    tmpfile = None
+    try:
+        command = [
+            "docker",
+            "run",
+            "-it",
+            image_tag,
+            "bash",
+            "-c",
+            f"{' '.join(command)} && bash",
```

Comment:
Initially on reading this I was worried about quoting corner cases here, but I guess since we only keep the run-task part of the command there's not actually much arbitrary input to take into account...  Consider using `shlex.join(command)` though?